include/fmt/format.h

```diff
@@ -3484,9 +3484,19 @@ template <typename Char> struct udl_arg {
   }
 };
 
+// Converts string literals to basic_string_view.
 template <typename Char, size_t N>
-FMT_CONSTEXPR basic_string_view<Char> literal_to_view(const Char (&s)[N]) {
-  return {s, N - 1};
+FMT_CONSTEXPR basic_string_view<Char> compile_string_to_view(
+    const Char (&s)[N]) {
+  // Remove trailing null character if needed. Won't be present if this is used
+  // with raw character array (i.e. not defined as a string).
+  return {s, N - (std::char_traits<Char>::to_int_type(s[N - 1]) == 0 ? 1 : 0)};
```

Comment:
Since you are extending this to more than string literals, please handle the case of N == 0 (and add a test).

Reply:
I added the bounds check, but as far as I can tell there's no way to declare a compile-time 0 length array in C++ without relying upon a non-standard compiler extension, so I was unable to add a test for it.